travis/publish.sh

```diff
@@ -104,6 +104,50 @@ safe_checkout_master() {
   fi
 }
 
+test_server() {
+  # Test Stackdriver Storage module with the latest version of Zipkin server.
+  temp_dir=$(mktemp -d)
+  pushd $temp_dir
+
+  # Download and unpack Zipkin Server
+  wget https://jitpack.io/com/github/openzipkin/zipkin/zipkin-server/master-SNAPSHOT/zipkin-server-master-SNAPSHOT-exec.jar
+
+  # Zipkin server JAR is a self-executable with headers that causes unzip to exit with status 1
+  # Fix the archive first.
+  zip -A zipkin-server-master-SNAPSHOT-exec.jar
+  unzip zipkin-server-master-SNAPSHOT-exec.jar -d zipkin
```

Comment:
I think a similar server test could be done by using maven project and adding the snapshot repo and dependency on the exec jar. Then you could just call ZipkinServer.main. A lot of the logic here like the massive wait-for-it would be simpler in Java.

What do you think of that approach?

Reply:
moving this back to the issue for more discussion. https://github.com/openzipkin/zipkin-gcp/issues/130#issuecomment-531521645

Reply:
I don't like the dependency on zipkin server approach. as it sends the wrong signal about supported integrations and also doesn't test the actual integration as we support it.

Reply:
for example those on gitter know we are plagued with ad-hoc questions about custom servers via dependencies, to the point of us wanting to refactor our entire README strategy. Please don't test via dependencies as A. people will get the wrong idea  B. it hides integration problems C. it doesn't follow our README.

Reply:
I dont know where my comment is, but I wholly do not like the idea of call ZipkinServer.main. I anti-like it, actively dislike it as it is doesn't actually test integration as in our README. We should test what we support

Reply:
apologies about the redundant comment.. github refresh wasn't refreshing. I'll delete the last one

Reply:
The main advantage would be if we want to run more tests than just the health, which would be much simpler I'm Java. But if just a health check gives us confidence then I think this is fine too.

Reply:
I understood the goal of this to be a smoke test, not a replacement for the IT stuff we do already. Ideally, we don't have to cover too many paths in this style.

Reply:
most of our problems have been related to things we now do in health check, so let's try this and re-evalutate if it is not pulling weight